packages/block-editor/src/components/block-rename/modal.js

```diff
@@ -17,14 +18,13 @@ import { speak } from '@wordpress/a11y';
  */
 import isEmptyString from './is-empty-string';
 
+// This component is only used in a non-WP environment.
+// Changes here should likely be duplicated in the `patterns` package as well.
```

Comment:
Could the patterns package component import and use this component to reduce the duplication? I imagine it'd mostly need a new generic prop or two (e.g. `helpText`) to make it work.